File: app/python_service/middlewares/handleError.py
from flask import jsonify # type: ignore

def handleError(code, error):
    """
    Handles error responses for the application.

    Args:
        error (Object): The error message to return.

    Returns:
        Response: A Flask JSON response with the error message and status code.
    """
    try:
        print(f"Error HTTP response: {code} - {error}")
        response = {
            "code": code,
            "msg": error
        }
        return jsonify(response), code
    except Exception as e: 
        return jsonify({}), 500 

File: app/server/src/middlewares/utils/handleError.js
/**
 * Handles error by printing to console in development env and builds and sends an error response
 * @param {Object} res - response object
 * @param {Object} err - error object
 */
export const handleError = (res = {}, err = {}) => {

  if (process.env.NODE_ENV === 'development') {
    // eslint-disable-next-line no-console
    console.log(err)
  }

  console.log('ERROR: ', err)

  if (err?.notice && err.notice != null) {
    // eslint-disable-next-line no-console
    console.log ('Notification Errors:', err.notice)
  }

  res.status(err?.code || 404).json({
    errors: {
      msg: err?.message
    }
  })
}

export default handleError